filebeat/beater/filebeat.go

```diff
@@ -56,46 +55,37 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 
 	var err error
 
-	// Init channels
-	fb.publisherChan = make(chan []*input.FileEvent, 1)
+	// Channel from harvesters to spooler
+	publisherChan := make(chan []*input.FileEvent, 1)
```

Comment:
uh, we don't even need to store crawler/publisher in fb struct. All components + shutdown order can become local to Run method.